examples/DataApi-Voting-poll/template/modules/data_l1/src/main/scala/com/my/currency/data_l1/Main.scala

```diff
@@ -26,35 +30,45 @@ object Main
     ClusterId(UUID.fromString("517c3a05-9219-471b-a54c-21b7d72f4ae5")),
     version = BuildInfo.version
   ) {
-  override def dataApplication: Option[BaseDataApplicationL1Service[IO]] = Option(BaseDataApplicationL1Service(new DataApplicationL1Service[IO, PollUpdate, State] {
-
-    override def serializeState(state: State): IO[Array[Byte]] = MainData.serializeState(state)
+  override def dataApplication: Option[BaseDataApplicationL1Service[IO]] = Option(BaseDataApplicationL1Service(new DataApplicationL1Service[IO, PollUpdate, VoteStateOnChain, VoteCalculatedState] {
+    override def validateUpdate(update: PollUpdate)(implicit context: L1NodeContext[IO]): IO[DataApplicationValidationErrorOr[Unit]] = shared_data.Main.validateUpdate(update)
 
-    override def deserializeState(bytes: Array[Byte]): IO[Either[Throwable, State]] = MainData.deserializeState(bytes)
+    override def validateData(state: DataState[VoteStateOnChain, VoteCalculatedState], updates: NonEmptyList[Signed[PollUpdate]])(implicit context: L1NodeContext[IO]): IO[DataApplicationValidationErrorOr[Unit]] = IO {
+      ().validNec
+    }
```

Comment:
```suggestion
   override def validateData(state: DataState[VoteStateOnChain, VoteCalculatedState], updates: NonEmptyList[Signed[PollUpdate]])(implicit context: L1NodeContext[IO]): IO[DataApplicationValidationErrorOr[Unit]] = ().validNec.pure[IO]
```
Also, it looks odd, esp. if this is going to serve as an example, that we are returning a validNec when the function is typed for `DataApplicationValidationErrorOr[Unit]` because it means the reader has to have knowledge of what that return type is defined as, similar to knowing an implementation of an interface in order to use it correctly.